common/app/templates/headerInlineJS/loadFonts.scala.js

```diff
@@ -1,87 +1,186 @@
-// Determine what type of font-hinting we want.
+/*
+bypass normal browser font-loading to avoid the FOIT. works like this:
 
-(function (window, document) {
-    if (!window.guardian.isModernBrowser) {
-        return false;
-    }
-    var fontHinting = function () {
-        var ua = navigator.userAgent,
-            windowsNT = /Windows NT (\d\.\d+)/.exec(ua),
-            hinting = 'Off',
-            version;
-
-        if (windowsNT) {
-            version = parseFloat(windowsNT[1], 10);
-
-            // For Windows XP-7
-            if (version >= 5.1 && version <= 6.1) {
-                if (/Chrome/.exec(ua) && version < 6.0) {
-                    // Chrome on windows XP wants auto-hinting
-                    hinting = 'Auto';
-                } else {
-                    // All others use cleartype
-                    hinting = 'Cleartype';
-                }
-            }
-        }
-        return hinting;
-    }();
+do you have fonts in localStorage?
+    yes – inject them (minimises 2nd layout as fonts are loaded before DOM is created)
+    no  – did the localStorage check go ok?
+        yes – ajax them in as JSON immediately, inject them and save them to localStorage
+        no  – load font files async using @@font-face
 
-    // Check to see if you should get webfonts, and then try to load them from localStorage if so
-    var cookieData = 'GU_fonts=off; domain=' + location.hostname + '; path=/';
+*/
 
-    function disableFonts() {
-        document.cookie = cookieData + '; max-age=' + (60 * 60 * 24 * 365);
-    }
+(function (window, document) {
+    var ua = navigator.userAgent;
 
-    function enableFonts() {
-        document.cookie = cookieData + '; expires=Thu, 01 Jan 1970 00:00:00 GMT';
-    }
+    // Determine what type of font-hinting we want.
+    var fontHinting = (function () {
+        var hinting;
+        function determineHinting() {
+            if (typeof hinting === 'undefined') {
+                try {
+                    var windowsNT = /Windows NT (\d\.\d+)/.exec(ua),
+                        version;
 
-    function fontsEnabled() {
-        return document.cookie.indexOf('GU_fonts=off') === -1;
-    }
+                    hinting = 'Off';
+                    if (windowsNT) {
+                        version = parseFloat(windowsNT[1], 10);
+
+                        // For Windows XP-7
+                        if (version >= 5.1 && version <= 6.1) {
+                            if (/Chrome/.exec(ua) && version < 6.0) {
+                                // Chrome on windows XP wants auto-hinting
+                                hinting = 'Auto';
+                            } else {
+                                // All others use cleartype
+                                hinting = 'Cleartype';
+                            }
+                        }
+                    }
+                } catch (e) {}
+            }
+            return hinting;
+        };
+        return determineHinting;
```

Comment:
does this need to return a function? seems like just returning the string would be fine